.github/workflows/linux-docker-compose.yml

```diff
@@ -0,0 +1,40 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Linux docker-compose
+on:
+  - push
+  - pull_request
```

Comment:
FYI:

I want to restrict by paths:

```yaml
on:
  push:
    paths:
      - docker-compose.yml
      - cpp/**/*
  pull_request:
    paths:
      - docker-compose.yml
      - cpp/**/*
```

But `**` isn't supported: https://help.github.com/en/articles/workflow-syntax-for-github-actions#onpushpull_requestpaths

Reply:
Can you enumerate all of the depths we have like they do in the example https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-matching-paths-in-nested-directories ?

Reply:
We can do but it'll be difficult to maintain. :<
It may be better that we use `ci/detect-changes.py` like we did on Travis CI and AppVeyor.

Reply:
The built-in way might be more efficient than the detect-changes script, and/or be more explicit that it's skipping a build. Worth exploring at least. 

Difficult to maintain, but how often do we add depth to the file tree in the cpp code? Could we preemptively do more than we currently have, like `cpp/*/*/*/*/*/*/*`?

Reply:
@kou this is just a draft, but I definitively had the intention to use `ci/detect-changes.py` to toggle whether we should build or pull the images.

We can use the same pattern to decide if we should run this image.

Reply:
@nealrichardson C++ case is simpler than others but other cases aren't so simple. For example, we need to maintain a list for C++ sources and R sources for R. We need to maintain a list for C++ sources, C GLib sources and Ruby sources for Ruby.

And we can't multiple `paths` set for one workflow file. We need to create one workflow file for one `paths`. It means that we can't use `matrix` to use other docker-compose image in this workflow file.

Reply:
@fsaintjacques I see.

Reply:
Not that there is a fundamental difference between setting pattern for GitHub Actions, and exiting the build with detect-changes.py
The former won't start the builds whereas the latter will start the builds in all cases and exit from it.

Reply:
I don't think it matters in practice, if the build starts and immediately exit (like travis does).